src/plugins/kibana/public/settings/sections/indices/add_data_steps/pattern_review_step/pattern_review_step.js

```diff
@@ -24,35 +26,51 @@ modules.get('apps/settings')
       controllerAs: 'reviewStep',
       bindToController: true,
       controller: function ($scope, Private) {
+        this.errors = [];
         if (_.isUndefined(this.indexPattern)) {
           this.indexPattern = {};
         }
 
         const knownFieldTypes = {};
         this.dateFields = [];
-        this.pipeline.model.processors.forEach((processor) => {
-          if (processor.typeId === 'geoip') {
-            const field = processor.targetField || 'geoip';
-            knownFieldTypes[field] = 'geo_point';
+        if (this.pipeline) {
+          this.pipeline.model.processors.forEach((processor) => {
+            if (processor.typeId === 'date') {
+              const field = processor.targetField || '@timestamp';
+              knownFieldTypes[field] = 'date';
+              this.dateFields.push(field);
+            }
+          });
+        }
+
+        const fields = {};
+        forEachField(this.sampleDoc, (value, fieldName) => {
+          let type = knownFieldTypes[fieldName] || typeof value;
+          if (isGeoPointObject(value)) {
+            type = 'geo_point';
+            knownFieldTypes[fieldName] = 'geo_point';
           }
-          else if (processor.typeId === 'date') {
-            const field = processor.targetField || '@timestamp';
-            knownFieldTypes[field] = 'date';
-            this.dateFields.push(field);
+
+          if (!_.isUndefined(fields[fieldName]) && (fields[fieldName].type !== type)) {
+            this.errors.push(`Error in field ${fieldName} - conflicting types '${fields[fieldName].type}' and '${type}'`);
+          }
+          else {
+            fields[fieldName] = {type, value};
           }
         });
 
+        const indexPatternFields = _(fields)
+        .map((field, fieldName) => {
+          return {name: fieldName, type: field.type};
+        })
+        .reject({type: 'object'})
```

Comment:
Excluding fields with a type of `object` excludes fields that are set to null by the processors. An example of this:
1. Use samples from [here](https://gist.githubusercontent.com/anonymous/1dbbe08809e1e643e854/raw/989dbebb9dd204d48f1a44f9f3bd8f29edc90480/sample%2520apache%2520logs).
2. Add a grok processor with this pattern: `%{COMMONAPACHELOG}`
3. The `rawrequest` field is created and set to `null` by the processor, but is not visible on the next step.